### ISSUE TITLE
can't resolve reference #/definitions/directiveConfigSchema from id #

### ISSUE THREAD

**Issue by ethanneff** [typescript-eslint/typescript-eslint#5538]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://github.com/typescript-eslint/typescript-eslint

### Repro Code

```typescript
any app
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

breaks on 
```
    "@typescript-eslint/eslint-plugin": "5.34.0",
    "@typescript-eslint/parser": "5.34.0",
```

### Actual Result

works on 
```
    "@typescript-eslint/eslint-plugin": "5.33.0",
    "@typescript-eslint/parser": "5.33.0",
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z` |

**Comment by JoshuaKGoldberg** [comment]
<img width="488" alt="You checked the I have searched for related issues and found none that matched my issue. entry in the checklist. A cursory issue search for this title determined that was a lie." src="https://user-images.githubusercontent.com/3335181/186501430-7f4c481a-f557-48b0-826c-74bb8434f4d5.png">

Dup of #5525. Next time, please use the search 🙂.
